src/core/symbology-ng/qgssymbollayerutils.cpp

```diff
@@ -3986,8 +4003,26 @@ double QgsSymbolLayerUtils::rescaleUom( double size, QgsUnitTypes::RenderUnit un
           scale = 1 / 0.28;
           roundToUnit = true;
           break;
-        // we don't have a good case for map units, as pixel values won't change based on zoom
-        default:
+        case QgsUnitTypes::RenderInches:
+          scale = 1 / 0.28 * 25.4;
+          roundToUnit = true;
+          break;
+        case QgsUnitTypes::RenderPoints:
+          scale = 90. /* dots per inch according to OGC SLD */ / 72. /* points per inch */;
+          roundToUnit = true;
+          break;
+        case QgsUnitTypes::RenderPixels:
+          // pixel is pixel
+          scale = 1;
+          break;
+        case QgsUnitTypes::RenderMapUnits:
+        case QgsUnitTypes::RenderMetersInMapUnits:
+          // already handed via uom
+          scale = 1;
+          break;
+        case QgsUnitTypes::RenderPercentage:
+        case QgsUnitTypes::RenderUnknownUnit:
+          // these do not make sense and should not really reach here
```

Comment:
Ho... I see, I totally agree with you! use the default value (scale = 1) and a big warning message in log or messagebar propably is enough.
Probably incorrect use of the api can be managed here:
https://github.com/aaime/QGIS/blob/61afededc1ca6d7c18dcfd64575538acd0a93cab/src/core/symbology-ng/qgssymbollayerutils.cpp#L3968
and in the default swtoch value to do travis happy.